usr/share/rear/build/default/980_verify_rootfs.sh

```diff
@@ -1,18 +1,50 @@
-# make sure that the rootfs contains a usable system
+# Test that the ROOTFS_DIR contains a usable system
+# i.e. test that the ReaR recovery system will be usable
+# to avoid issues like https://github.com/rear/rear/issues/1494
 
-# In case the filesystem is mounted 'noexec' we skip a chroot bash test
-tmp_filesystem=$(filesystem_name $ROOTFS_DIR)
+# In case the filesystem that contains the ROOTFS_DIR is mounted 'noexec' we must skip the 'chroot' tests.
+# The filesystem_name function in linux-functions.sh returns the mountpoint (not a filesystem name like 'ext4'):
+rootfs_dir_fs_mountpoint=$( filesystem_name $ROOTFS_DIR )
 
-if grep -qE '^\S+ '$tmp_filesystem' \S+ \S*\bnoexec\b\S* ' /proc/mounts; then
-	LogPrint "WARNING: Filesystem $tmp_filesystem is mounted 'noexec', aborting chroot bash test"
-	Log "The above error means that we cannot test our chrooted environment because
+if grep -qE '^\S+ '$rootfs_dir_fs_mountpoint' \S+ \S*\bnoexec\b\S* ' /proc/mounts ; then
+    LogPrint "Cannot test if the ReaR recovery system is usable because $rootfs_dir_fs_mountpoint is mounted 'noexec'"
+    Log "We cannot run tests inside the ReaR recovery system by using 'chroot' because
 we need the 'exec' option set to the filesystem. One way to achieve this is
-by doing: mount -o remount,exec $tmp_filesystem"
-	return
+by doing: mount -o remount,exec $rootfs_dir_fs_mountpoint"
+    return
 fi
 
-# The chroot bash test ensures that we have a working bash on our rescue image
-if ! chroot $ROOTFS_DIR bash -c true ; then
-	KEEP_BUILD_DIR=1
-	BugError "ROOTFS_DIR '$ROOTFS_DIR' is broken, chroot bash test failed."
+# The bash test ensures that we have a working bash in the ReaR recovery system:
+if ! chroot $ROOTFS_DIR /bin/bash -c true ; then
+    KEEP_BUILD_DIR=1
+    BugError "ReaR recovery system in '$ROOTFS_DIR' is broken: 'bash -c true' failed"
 fi
+
+# The ldd test ensures that for dynamically linked executables the required libraries are there.
+# The ldd test runs after the bash test because /bin/ldd is a bash script.
+# First test is 'ldd /bin/bash' to ensure 'ldd' works:
+if ! chroot $ROOTFS_DIR /bin/ldd /bin/bash 1>&2 ; then
+    KEEP_BUILD_DIR=1
+    BugError "ReaR recovery system in '$ROOTFS_DIR' is broken: 'ldd /bin/bash' failed"
+fi
+# Now test each binary (except links) with ldd and look for 'not found' libraries.
+# In case of 'not found' libraries for dynamically linked executables ldd returns zero exit code.
+# When running ldd for a file that is 'not a dynamic executable' ldd returns non-zero exit code.
+local binary=""
+local broken_binaries=""
+pushd $ROOTFS_DIR 1>&2
+for binary in $( find bin/ -type f ) ; do
```

Comment:
This doesn't catch all the binaries copied via `COPY_AS_IS` into other paths.

Instead of the `pushd` and `popd` I would suggest to use `find $ROOTFS_DIR -type f -executable -printf "/%P\n"`, it also catches all other binaries.

Reply:
Thanks for the improvement.
<pre>
find $ROOTFS_DIR -type f -executable -printf '/%P\n'
</pre>
finds all binaries and also all libraries and the latter causes
new isues because some libraries need other libraries that
are not there...